### PR TITLE
Dockerfile Smoke Tests

### LINKED ISSUE #103: Dockerfile smoke tests
## Summary
Create a new CI workflow that runs a smoke test job when a PR is open.

## Acceptance Criteria
- [ ] New CI `check_smoke_tests.yml` workflow

## Details / Tasks
The smoke test should roughly do the following, but more steps can be added if identified during the development process
- start up a database
- build the Dockerfile
- run the RL service on the docker container
- send a series (~4 or so) of curl requests to the /link endpoint, and verify results

## Background / Context
Smoke tests are a good way to verify basic functionality of the service is working.  Much of the details of the API are tested within unit tests, but we don't have a way right now to verify that the Dockerfile image still builds as expected.  Smoke tests should help with that.

### PR DESCRIPTION
**Pull request by cbrinson-rise8**
## Description
Create a new CI workflow that runs a smoke test job when a PR is open.

## Related Issues
closes #103 

## Additional Notes
The smoke test should roughly do the following, but more steps can be added if identified during the development process
- start up a database
- build the Dockerfile
- run the RL service on the docker container
- send a series (~4 or so) of curl requests to the /link endpoint, and verify results

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/check_smoke_tests.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Set up Docker
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y docker-compose
+
+    - name: Start services with docker-compose
+      run: docker-compose up -d
```
Comment (ericbuckley) [inline .github/workflows/check_smoke_tests.yml:26]: The `compose.yml` file is currently configured for development use, the main differences being it will run jaeger by default and override the default CMD to wrap the web server in an opentelemetry call.  Neither of these will likely be used in production environments.  Did you consider just using docker on its own to run the API?  The only dependency would be a database like postgres.

File: .github/workflows/check_smoke_tests.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        JSON_BODY_1='{"record": {"birth_date": "2053-11-07", "sex": "M", "mrn": "1234567890", "name":[{"family":"Shepard", "given":["John"]}]}}'
+        JSON_BODY_2='{"algorithm": "dibbs-enhanced", "record": {"birth_date": "2000-12-06", "sex": "M", "mrn": "9876543210", "name":[{"family":"Smith", "given":["William"]}]}}'
+
+        #basic tests
+        RESPONSE_1=$(curl -s -X POST http://localhost:8080/link \
+        -d "$JSON_BODY_1" \
+        -H "Content-Type: application/json")
+
+        echo "Response: $RESPONSE_1"
+        echo "$RESPONSE_1" | grep -q '"is_match":\s*false'
```
Comment (ericbuckley) [inline .github/workflows/check_smoke_tests.yml:55]: Cool use of grep.  I believe `jq` is also available by default on ubuntu-latest if you wanted to use that.

File: .github/workflows/check_smoke_tests.yml (hunk 1 above)
Comment (cbrinson-rise8) [reply .github/workflows/check_smoke_tests.yml:26]: Yeah I could definitly change it to use that. No real reason I did it this way other than it was the first thing I thought of. I'll take a look at just using the dockerfile instead

File: .github/workflows/check_smoke_tests.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Start PostgreSQL Database
```
Comment (ericbuckley) [inline .github/workflows/check_smoke_tests.yml:20]: @cbrinson-rise8 [services](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idservices) make dependencies like this a bit easier to manage.  For instance, take a look at this [postgres service we create for unit tests](https://github.com/CDCgov/RecordLinker/blob/main/.github/workflows/check_unit_tests.yml#L17-L23).  We can then use that service in any of our steps by [referencing the hostname that matches the service name](https://github.com/CDCgov/RecordLinker/blob/main/.github/workflows/check_unit_tests.yml#L73) (in this case, "postgres").

File: src/recordlinker/linking/algorithm_service.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def default_algorithm(session: orm.Session) -> models.Algorithm:
     query = select(models.Algorithm).where(models.Algorithm.is_default == True)  # noqa: E712
     res = session.scalars(query).all()
     if not res:
-        raise ValueError("No default algorithm found in the database.")
+        return None
```
Comment (ericbuckley) [inline src/recordlinker/linking/algorithm_service.py:38]: Would you mind also adding a test to the test_link_router that shows when we call the endpoints and no default is defined it results in a 422 and not a 500?

Comment (cbrinson-rise8) [reply src/recordlinker/linking/algorithm_service.py:38]: Added

**Review by ericbuckley — APPROVED** [review]
@cbrinson-rise8 this is great, thanks so much for working on this, I think its going to help us catch some future bugs early!
